blockchain/chain.go

```diff
@@ -305,17 +305,31 @@ type StakeVersions struct {
 // GetStakeVersions returns a cooked array of StakeVersions.  We do this in
 // order to not bloat memory by returning raw blocks.
 func (b *BlockChain) GetStakeVersions(hash *chainhash.Hash, count int32) ([]StakeVersions, error) {
+	// Nothing to do if no count requested.
+	if count == 0 {
+		return nil, nil
+	}
+
 	b.chainLock.Lock()
 	defer b.chainLock.Unlock()
 
+	// Enforce sane count bounds.
+	if count < 0 {
```

Comment:
Well this one can, but the next one (`bestNode`) can't. Good observation. Updated.